k8s/client.go

```diff
@@ -16,27 +25,79 @@ type Client interface {
 	WatchIngresses(Watcher) error
 }
 
-type noopClient struct {
+type impl struct {
+	baseURL string
+	caCert  []byte
+	token   string
+	client  *http.Client
 }
 
-// NewClient creates a new K8 Client
-func NewClient() Client {
-	return &noopClient{}
-}
+// New creates a client for the kubernetes apiserver.
+func New(apiServerURL string, caCert []byte, token string) (Client, error) {
+	parsedURL, err := url.Parse(apiServerURL)
+	if err != nil {
+		return nil, fmt.Errorf("invalid url %s: %v", apiServerURL, err)
+	}
+	baseURL := strings.TrimSuffix(parsedURL.String(), "/")
+
+	pool := x509.NewCertPool()
+	pool.AppendCertsFromPEM(caCert)
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{RootCAs: pool},
+	}
+	client := &http.Client{Transport: tr}
+
+	log.Debugf("Constructing client with url: %s, token: %s, caCert: %v",
```

Comment:
It's really noisy, the cert and tokens are big. Like it prints 10-15 lines of stuff. We could have a single info saying it's creating the client? But I'm not sure what that offers, no external interactions happen until it tries to do a request.